apps/ensadmin-next/src/components/connections/connection-selector.tsx

```diff
@@ -136,81 +138,87 @@ export function ConnectionSelector() {
                       </button>
                     )}
                   </div>
-                </div>
+                </DropdownMenuItem>
               ))
             )}
-          </div>
 
-          {/* Add New Connection */}
-          {isAdding ? (
-            <div className="px-3 py-2 space-y-2">
-              <input
-                type="url"
-                placeholder="Enter URL"
-                value={newUrl}
-                onChange={(e) => setNewUrl(e.target.value)}
-                disabled={addConnection.isPending}
-                className={cn(
-                  "w-full px-2 py-1 text-sm font-mono rounded border bg-background",
-                  addConnection.isError ? "border-destructive" : "border-input",
-                  addConnection.isPending && "opacity-50",
-                )}
-              />
-              {addConnection.isError && (
-                <p className="text-xs text-destructive">
-                  {addConnection.error instanceof Error
-                    ? addConnection.error.message
-                    : "Failed to add connection"}
-                </p>
-              )}
-              <div className="flex justify-end gap-2">
-                <button
-                  onClick={() => {
-                    setIsAdding(false);
-                    setNewUrl("");
-                    addConnection.reset();
-                  }}
-                  disabled={addConnection.isPending}
-                  className="px-2 py-1 text-xs text-muted-foreground hover:text-foreground disabled:opacity-50"
-                >
-                  Cancel
-                </button>
-                <button
-                  onClick={handleAdd}
+            <DropdownMenuSeparator />
+
+            {isAdding ? (
+              <div className="px-3 py-2 space-y-2">
```

Comment:
I was thinking of changing this one to `form` tag, so we could add `onSubmit` and `onReset` callbacks, and let the "Cancel" button be of `type="reset"` and "Add" button be of `type="submit"`.

With that change, we'd get a native form UX, where hitting `<enter>` while focus within any element of a form, submits the form right away.

Reply:
Yeah, I didn't touch the functionality here. I want to move this into a new complete slideover component in a follow up PR.